qualtran/serialization/bloq.py

```diff
@@ -345,8 +311,9 @@ def _bloq_to_proto(bloq: Bloq, *, bloq_to_idx: Dict[Bloq, int]) -> bloq_pb2.Bloq
     except (DecomposeTypeError, DecomposeNotImplementedError, TypeError):
         t_complexity = None
 
+    name = bloq.__module__ + "." + bloq.__class__.__qualname__
```

Comment:
how will we handle moving bloqs w/ backwards compatibility. You want to always serialize the latest fully-qualified name and then have multiple entries in resolverdict?

Reply:
That's one way. Or we can introduce a `_qualtran_namespace_` protocol, similar to Cirq's `_json_namespace_`, and use that instead of explicit fully qualified names. This line would then change to `bloq._namespace_() + "." + bloq.__class__.__qualname__`. 

Adding a new protocol to the Bloq class can be done in a follow-up PR if needed.